### ISSUE TITLE
ArrayIndexOutOfBoundsException: 8 at InstrumentingMethodVisitor.java:1337

### ISSUE THREAD

**Issue by szhem** [btraceio/btrace#367]
Hi there,

Trying to trace some Apache Spark code I've faced with the following issue
```
java.lang.ArrayIndexOutOfBoundsException: 8
	at com.sun.btrace.runtime.InstrumentingMethodVisitor.computeFrameLocals(InstrumentingMethodVisitor.java:1337)
	at com.sun.btrace.runtime.InstrumentingMethodVisitor.visitFrame(InstrumentingMethodVisitor.java:427)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.a(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.b(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.btrace.runtime.BTraceClassReader.accept(BTraceClassReader.java:174)
	at com.sun.btrace.org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.btrace.runtime.BTraceClassReader.accept(BTraceClassReader.java:183)
	at com.sun.btrace.runtime.InstrumentUtils.accept(InstrumentUtils.java:177)
	at com.sun.btrace.runtime.InstrumentUtils.accept(InstrumentUtils.java:171)
	at com.sun.btrace.runtime.BTraceClassWriter.instrument(BTraceClassWriter.java:82)
	at com.sun.btrace.runtime.BTraceTransformer.transform(BTraceTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:45)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.CoalescedRDD$$anonfun$compute$1.apply(CoalescedRDD.scala:100)
	at org.apache.spark.rdd.CoalescedRDD$$anonfun$compute$1.apply(CoalescedRDD.scala:99)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:438)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.internal.io.SparkHadoopMapReduceWriter$$anonfun$4.apply(SparkHadoopMapReduceWriter.scala:150)
	at org.apache.spark.internal.io.SparkHadoopMapReduceWriter$$anonfun$4.apply(SparkHadoopMapReduceWriter.scala:148)
	at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1374)
	at org.apache.spark.internal.io.SparkHadoopMapReduceWriter$.org$apache$spark$internal$io$SparkHadoopMapReduceWriter$$executeTask(SparkHadoopMapReduceWriter.scala:163)
	at org.apache.spark.internal.io.SparkHadoopMapReduceWriter$$anonfun$3.apply(SparkHadoopMapReduceWriter.scala:93)
	at org.apache.spark.internal.io.SparkHadoopMapReduceWriter$$anonfun$3.apply(SparkHadoopMapReduceWriter.scala:88)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

and here is the tracing snippet itself
```java
    @OnMethod(
        clazz = "org.apache.spark.storage.ShuffleBlockFetcherIterator",
        method = "next",
        location = @Location(Kind.ENTRY)
    )
    public static void shuffleBlockFetcherIterator_next_exit(
        @Self ShuffleBlockFetcherIterator obj,
        @ProbeMethodName String method
    ) {
        info(obj.getClass().getSimpleName(), method);
    }
```

Could you please suggest the ways to fix or workaround it?

**Comment by jbachorik** [comment]
Most probably related to #333. Working on the fix now.

**Comment by jbachorik** [comment]
Hi, I addressed this issue in develop branch. Can you, please, verify?

**Comment by szhem** [comment]
Hi @jbachorik 

Sorry for the delayed response.
BTrace built from latest master hasn't helped.

Here are the argument passed to JVM
```shell
-javaagent:./btrace-agent.jar=script=./btrace/ExecutorProbe3.class,stdout=true,trusted=true,debug=true
```
... and here is the output
```
btrace DEBUG: parsed command line arguments
btrace DEBUG: debugMode is true
btrace DEBUG: trustedMode is true
btrace DEBUG: stdout is true
btrace DEBUG: initial script is /home/szhem/Projects/spark/btrace/ExecutorProbe3.class
btrace DEBUG: loading BTrace class
### BTrace Log: 11/26/18 9:53 PM

btrace DEBUG: loaded 'btrace.ExecutorProbe3' successfully
btrace DEBUG: creating BTraceRuntime instance for btrace.ExecutorProbe3
btrace DEBUG: created BTraceRuntime instance for btrace.ExecutorProbe3
btrace DEBUG: sending Okay command
btrace DEBUG: client btrace.ExecutorProbe3: got com.sun.btrace.comm.OkayCommand@6504e3b2
btrace DEBUG: about to defineClass btrace/ExecutorProbe3
btrace DEBUG: defineClass succeeded for btrace.ExecutorProbe3
btrace DEBUG: new Client created com.sun.btrace.agent.FileClient@5f7dc0eb
btrace DEBUG: retransforming loaded classes
btrace DEBUG: filtering loaded classes
btrace DEBUG: noServer is true, server not started
btrace DEBUG: Agent init took: 969926192ns
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$1
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$2
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$3
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$4
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MemberName$Factory
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleStatics
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleStatics$1
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeStaticBooleanFieldAccessorImpl
btrace DEBUG: client btrace.ExecutorProbe3: got com.sun.btrace.comm.OkayCommand@2b8b6041
btrace DEBUG: skipping transform for BTrace class sun/security/jca/Providers
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderList
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderConfig
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderList$3
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderList$1
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderList$2
btrace DEBUG: skipping transform for BTrace class sun/security/jca/ProviderConfig$2
btrace DEBUG: skipping transform for BTrace class sun/security/provider/Sun
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SunEntries
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SunEntries$1
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$Variant
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$1
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$2
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$RandomIO
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$Blocking
btrace DEBUG: skipping transform for BTrace class sun/security/provider/NativePRNG$NonBlocking
btrace DEBUG: skipping transform for BTrace class sun/security/jca/GetInstance
btrace DEBUG: skipping transform for BTrace class sun/security/jca/GetInstance$Instance
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SecureRandom
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SHA
btrace DEBUG: skipping transform for BTrace class sun/security/provider/DigestBase
btrace DEBUG: skipping transform for BTrace class sun/security/provider/ByteArrayAccess
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/FormalTypeParameter
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/ClassSignature
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/Signature
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/scope/MethodScope
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/repository/MethodRepository
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/repository/ConstructorRepository
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/BottomSignature
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/Wildcard
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/MethodTypeSignature
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/reflectiveObjects/WildcardTypeImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/reflectiveObjects/LazyReflectiveObjectGenerator
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/reflectiveObjects/ParameterizedTypeImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/reflectiveObjects/TypeVariableImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeQualifiedObjectFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeQualifiedFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeObjectFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class java/lang/StringIndexOutOfBoundsException
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/JavaxSecurityAuthKerberosAccess
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/KerberosSecrets
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/Config
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/internal/Krb5
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/KrbException
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/Config$FileExistsAction
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/Config$1
btrace DEBUG: skipping transform for BTrace class sun/security/krb5/Config$2
WARN  | NativeCodeLoader | ScalaTest-run-running-MeetupSpec | Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaMetafactory
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandles$Lookup
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodType$ConcurrentWeakInternSet
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodTypeForm
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/Invokers
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodType$ConcurrentWeakInternSet$WeakEntry
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandles
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaForm$BasicType
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaForm$Name
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaForm$NamedFunction
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/SimpleMethodHandle
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$SpeciesData
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Factory
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Species_L
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/DirectMethodHandle$Lazy
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InvokerBytecodeGenerator
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$Intrinsic
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InvokerBytecodeGenerator$2
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InvokerBytecodeGenerator$CpPatch
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/DirectMethodHandle$Accessor
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$AsVarargsCollector
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/DelegatingMethodHandle
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/WrongMethodTypeException
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$Lazy
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleImpl$IntrinsicMethodHandle
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaFormEditor
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaFormEditor$Transform$Kind
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaFormEditor$Transform
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/LambdaFormBuffer
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$SpeciesData$1
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Factory$1
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Species_LL
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeStaticObjectFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Species_L3
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/BoundMethodHandle$Species_L4
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InnerClassLambdaMetafactory
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/AbstractValidatingLambdaMetafactory
btrace DEBUG: skipping transform for BTrace class sun/security/util/SecurityConstants
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InfoFromMemberName
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/MethodHandleInfo
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InnerClassLambdaMetafactory$ForwardingMethodGenerator
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/TypeConvertingMethodAdapter
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/InnerClassLambdaMetafactory$1
btrace DEBUG: skipping transform for BTrace class java/lang/invoke/DirectMethodHandle$Special
WARN  | SparkConf | ScalaTest-run-running-MeetupSpec | In Spark 1.0 and later spark.local.dir will be overridden by the value set by the cluster manager (via SPARK_LOCAL_DIRS in mesos/standalone and LOCAL_DIRS in YARN).
WARN  | Utils | ScalaTest-run-running-MeetupSpec | Your hostname, szhem-vb-2 resolves to a loopback address: 127.0.1.1; using 10.0.2.15 instead (on interface eth0)
WARN  | Utils | ScalaTest-run-running-MeetupSpec | Set SPARK_LOCAL_IP if you need to bind to another address
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeLongFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/security/action/GetIntegerAction
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeQualifiedStaticIntegerFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/BootstrapConstructorAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/scope/ConstructorScope
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/VoidDescriptor
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/repository/FieldRepository
btrace DEBUG: skipping transform for BTrace class sun/reflect/UnsafeQualifiedStaticLongFieldAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/SerializationConstructorAccessorImpl
btrace DEBUG: skipping transform for BTrace class sun/reflect/generics/tree/TypeVariableSignature
btrace DEBUG: skipping transform for BTrace class sun/reflect/misc/MethodUtil
btrace DEBUG: skipping transform for BTrace class sun/reflect/misc/MethodUtil$1
btrace DEBUG: skipping transform for BTrace class sun/security/provider/MD5
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SHA2$SHA256
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SHA2
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SHA5$SHA512
btrace DEBUG: skipping transform for BTrace class sun/security/provider/SHA5
[Stage 0:>                                                          (0 + 0) / 5]***
***
***
***
***
***
btrace DEBUG: java.lang.ArrayIndexOutOfBoundsException: 8
java.lang.ArrayIndexOutOfBoundsException: 8
	at com.sun.btrace.runtime.InstrumentingMethodVisitor.computeFrameLocals(InstrumentingMethodVisitor.java:1337)
	at com.sun.btrace.runtime.InstrumentingMethodVisitor.visitFrame(InstrumentingMethodVisitor.java:427)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.MethodVisitor.visitFrame(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.a(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.b(Unknown Source)
	at com.sun.btrace.org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.btrace.runtime.BTraceClassReader.accept(BTraceClassReader.java:174)
	at com.sun.btrace.org.objectweb.asm.ClassReader.accept(Unknown Source)
	at com.sun.btrace.runtime.BTraceClassReader.accept(BTraceClassReader.java:183)
	at com.sun.btrace.runtime.InstrumentUtils.accept(InstrumentUtils.java:177)
	at com.sun.btrace.runtime.InstrumentUtils.accept(InstrumentUtils.java:171)
	at com.sun.btrace.runtime.BTraceClassWriter.instrument(BTraceClassWriter.java:82)
	at com.sun.btrace.runtime.BTraceTransformer.transform(BTraceTransformer.java:184)
	at sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:428)
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:45)
	at org.apache.spark.rdd.CoGroupedRDD$$anonfun$compute$2.apply(CoGroupedRDD.scala:148)
	at org.apache.spark.rdd.CoGroupedRDD$$anonfun$compute$2.apply(CoGroupedRDD.scala:137)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at org.apache.spark.rdd.CoGroupedRDD.compute(CoGroupedRDD.scala:137)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:338)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

btrace DEBUG: client btrace.ExecutorProbe3: got com.sun.btrace.comm.ExitCommand@254c8b40
btrace DEBUG: onExit:
btrace DEBUG: cleaning up transformers
btrace DEBUG: onExit: removing transformer for btrace.ExecutorProbe3
btrace DEBUG: removing instrumentation
btrace DEBUG: retransforming loaded classes
btrace DEBUG: filtering loaded classes
btrace DEBUG: candidate class org.apache.spark.storage.ShuffleBlockFetcherIterator added
btrace DEBUG: client btrace.ExecutorProbe3: got com.sun.btrace.comm.RetransformationStartNotification@5298ea21
btrace DEBUG: calling retransformClasses (1 classes to be retransformed)
btrace DEBUG: Attempting to retransform class: org.apache.spark.storage.ShuffleBlockFetcherIterator
btrace DEBUG: closing all I/O
```
